Source/Diagnostics/FullDiagnostics.cpp

```diff
@@ -351,6 +351,9 @@ FullDiagnostics::InitializeFieldFunctors (int lev)
     // Clear any pre-existing vector to release stored data.
     m_all_field_functors[lev].clear();
 
+    // Species index to loop over species that dump rho per species
+    int is_dump_rho = 0;
```

Comment:
This shared state might be a bit too implicit to quickly follow and keep track of. Can we derive the final value in `is_dump_rho` from something else, e.g. `m_rho_per_species_index.size()` or the count of non-zero entries in `m_rho_per_species_index` ([`std::count_if`](https://en.cppreference.com/w/cpp/algorithm/count))?

Reply:
Clarified: I overlooked that this a temporary variable, no member variable.